app/script/auth/page/Login.js

```diff
@@ -0,0 +1,282 @@
+import * as ConversationAction from '../module/action/ConversationAction';
+import ValidationError from '../module/action/ValidationError';
+import {loginStrings} from '../../strings';
+import RuntimeUtil from '../util/RuntimeUtil';
+import * as URLUtil from '../util/urlUtil';
+import AppAlreadyOpen from '../component/AppAlreadyOpen';
+import BackendError from '../module/action/BackendError';
+import {Redirect} from 'react-router';
+
+class Login extends React.PureComponent {
+  inputs = {};
+
+  state = {
+    conversationCode: null,
+    conversationKey: null,
+    email: '',
+    isValidLink: true,
+    loginError: null,
+    password: '',
+    persist: true,
+    validInputs: {
+      email: true,
+      password: true,
+    },
+    validationErrors: [],
+  };
+
+  readAndUpdateParamsFromUrl = (nextProps = this.props) => {
+    const conversationCode = nextProps.match.params.conversationCode;
+    const conversationKey = nextProps.match.params.conversationKey;
+
+    const keyAndCodeExistent = conversationKey && conversationCode;
+    const keyChanged = conversationKey !== this.state.conversationKey;
+    const codeChanged = conversationCode !== this.state.conversationCode;
+    const keyOrCodeChanged = keyChanged || codeChanged;
+    if (keyAndCodeExistent && keyOrCodeChanged) {
+      Promise.resolve()
+        .then(() => {
+          this.setState((state, props) => ({
+            ...state,
+            conversationCode,
+            conversationKey,
+            isValidLink: true,
+          }));
+        })
+        .then(() => this.props.doCheckConversationCode(conversationKey, conversationCode))
+        .catch(error => {
+          this.setState((state, props) => ({
+            ...state,
+            isValidLink: false,
+          }));
+        });
+    }
+  };
+
+  componentDidMount = () => this.readAndUpdateParamsFromUrl();
+
+  componentWillReceiveProps = nextProps => this.readAndUpdateParamsFromUrl(nextProps);
+
+  handleSubmit = event => {
+    event.preventDefault();
+    if (this.props.isFetching) {
+      return;
+    }
+    const validationErrors = [];
+    const validInputs = this.state.validInputs;
+    for (const inputKey of Object.keys(this.inputs)) {
+      const currentInput = this.inputs[inputKey];
+      if (!currentInput.checkValidity()) {
+        validationErrors.push(ValidationError.handleValidationState(currentInput.name, currentInput.validity));
+      }
+      validInputs[inputKey] = currentInput.validity.valid;
+    }
+    this.setState({validInputs, validationErrors});
+    return Promise.resolve(validationErrors)
+      .then(errors => {
+        if (errors.length) {
+          throw errors[0];
+        }
+      })
+      .then(() => {
+        const {email, password, persist} = this.state;
+        const login = {password, persist};
+        if (email.includes('@')) {
+          login.email = email;
+        } else {
+          login.handle = email.replace('@', '');
+        }
```

Comment:
This is the spot to readd the secret phone login with password functionality we previously had.